### PR TITLE
Coordinating agent

### PR DESCRIPTION
**Pull request by SytzeAndr**

### CODE REVIEW

**Comment by SytzeAndr** [comment]
Observations are split up such that only turbine data from nearby turbines is passed on.

A threshold can be set in the config yaml file.

Currently this is only functional for the naive agent.

File: code/agent/multi/multi_agent.py [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+        pass
+
+
+def distance_matrix(turbine_layout):
+    turbine_coordinates = list(zip(turbine_layout["x"], turbine_layout["y"]))
+    return scipy.spatial.distance.cdist(turbine_coordinates, turbine_coordinates)
+
+
+def is_within_distance_mapping(distance_matrix, threshold):
+    return list(map(lambda row: [i for i, x in enumerate(row) if x < threshold], distance_matrix))
```
Comment (Alfa-R) [inline code/agent/multi/multi_agent.py:96]: should be <= threshold

File: code/agent/multi/multi_agent.py [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+
+
+if __name__ == "__main__":
+    turbine_x = [0, 1, 2, 5]
+    turbine_y = [5, 0, 2, 8]
+    turbine_coordinates = list(zip(turbine_x, turbine_y))
+    dist = distance_matrix(turbine_coordinates)
+    within_distance = is_within_distance_mapping(dist, 4)
```
Comment (SytzeAndr) [inline code/agent/multi/multi_agent.py:104]: delete

File: code/agent/multi/multi_agent.py [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+
+        # compute for each turbine what other turbines are within distance and store in matrix
+        self.turbines_in_range_list = is_within_distance_mapping(
+            distance_matrix(turbine_layout),
+            distance_threshold
+        )
+
+        # create an agent for each turbine
+        def create_agent(row):
+            state_space_size_per_agent = 3
```
Comment (Alfa-R) [inline code/agent/multi/multi_agent.py:22]: 3 should not be hardcoded

**Pull request closed without merging**